src/datastore/local-store/repos/users.repo.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@nestjs/common';
+
+import { AppConflictError, AppNotFoundError } from '~/common/errors';
+import { CreateUserDto } from '~/users/dto/create-user.dto';
+import { UserModel } from '~/users/model/user.model';
+import { UsersRepo } from '~/users/repo/user.repo';
+import { defaultUser } from '~/users/user.consts';
+
+@Injectable()
+export class LocalUserRepo implements UsersRepo {
+  private users: Record<string, UserModel> = { [defaultUser.id]: defaultUser };
+  private _nextId = 1;
+
+  public async createUser(params: CreateUserDto): Promise<UserModel> {
+    const { name } = params;
+    const existingUser = this.getUserByName(name);
+    if (existingUser) {
+      throw new AppConflictError(name, UsersRepo.type);
+    }
+
+    const id = this.nextId();
```

Comment:
I kind of went back and forth on this. I am fine switching to a uuid v4 style identifier for local as well, but its more intended as a dev config, so the sequence isn't as much of a concern.

Reply:
I did have Postgres using serial for a bit, but that can leak user growth metrics to competitors for some users.

Luckily local is ephemeral so we can change this whenever and not worry about data migrations though

Reply:
Node v14.16 introduces a native UUID method, but we claim to support just node v14+ right now.

It won't be a breaking change to use UUID in the future and we can avoid a dependency by waiting until we officially stop supporting older node versions.

Does that make sense to you? This is easy to change since we never have to deal with old data here. As a dev I think its nice to get an implicit hint of what type of DB is being used, and a consumer should only treat the ID as an opaque string.